types.go

```diff
@@ -15,54 +15,62 @@ import (
 
 const defaultGradientAbsTol = 1e-6
 
-// EvaluationType is used by a Method to specify the objective-function
-// information needed at an x location.
-type EvaluationType uint
+// Operation represents the set of operations requested by Method at each
+// iteration. It is a bitmap of *Iteration and *Evaluation constants.
```

Comment:
That notation is confusing. I originally read it as "Pointer to Iteration type" and "Pointer to evaluation type". 

We need to comment that a Method may return MajorIteration xor NoOperation xor a combination of the evaluations. A method may never return Init and Post. It's probably commented in Method, but I don't think it hurts to have it in multiple places in this case.